designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ScenarioActivityAuditLog.scala

```diff
@@ -0,0 +1,221 @@
+      user: LoggedUser
+  ): Unit =
+    logger.info(
+      withPrefix(ScenarioId(processId.value), processVersion.map(ScenarioVersionId.from), user.username)(
+        s"Immediate scenario action [actionName=${actionName.value},actionId=${processActionId.value}]"
+      )
+    )
+
+  def onScenarioActionStarted(
+      processActionId: ProcessActionId,
+      processId: ProcessId,
+      actionName: ScenarioActionName,
+      processVersion: Option[VersionId],
+      user: LoggedUser
+  ): Unit =
+    logger.info(
+      withPrefix(ScenarioId(processId.value), processVersion.map(ScenarioVersionId.from), user.username)(
+        s"Scenario action [actionName=${actionName.value},actionId=${processActionId.value}] started"
+      )
+    )
+
+  def onScenarioActionFinishedWithSuccess(
+      processActionId: ProcessActionId,
+      processId: ProcessId,
+      actionName: ScenarioActionName,
+      processVersion: Option[VersionId],
+      comment: Option[String],
+      user: LoggedUser
+  ): Unit = {
+    val commentValue = comment match {
+      case Some(content) => s"comment [$content]"
+      case None          => "without comment"
+    }
+    logger.info(
+      withPrefix(ScenarioId(processId.value), processVersion.map(ScenarioVersionId.from), user.username)(
+        s"Scenario action [actionName=${actionName.value},actionId=${processActionId.value}] finished with success and $commentValue "
+      )
+    )
+  }
+
+  def onScenarioActionFinishedWithFailure(
+      processActionId: ProcessActionId,
+      processId: ProcessId,
+      actionName: ScenarioActionName,
+      processVersion: Option[VersionId],
+      comment: Option[String],
+      failureMessage: String,
+      user: LoggedUser
+  ): Unit = {
+    val commentValue = comment match {
+      case Some(content) => s"with comment [$content]"
+      case None          => "without comment"
+    }
+    logger.info(
+      withPrefix(ScenarioId(processId.value), processVersion.map(ScenarioVersionId.from), user.username)(
+        s"Scenario action [actionName=${actionName.value},actionId=${processActionId.value}] finished with failure [$failureMessage] $commentValue"
+      )
+    )
+  }
+
+  def onScenarioActionRemoved(
+      processActionId: ProcessActionId,
+      processId: ProcessId,
+      processVersion: Option[VersionId],
+      user: LoggedUser
+  ): Unit = {
+    logger.info(
+      withPrefix(ScenarioId(processId.value), processVersion.map(ScenarioVersionId.from), user.username)(
+        s"Scenario action [actionId=${processActionId.value}] removed"
+      )
+    )
+  }
+
+  private def withPrefix(scenarioId: ScenarioId, scenarioVersionId: Option[ScenarioVersionId], username: String)(
+      log: String
+  ) = {
+    val scenarioIdValue        = scenarioId.value
+    val scenarioVersionIdValue = scenarioVersionId.map(_.value.toString).getOrElse("none")
+    s"[SCENARIO_AUDIT][scenarioId=$scenarioIdValue][version=$scenarioVersionIdValue][user=$username] $log"
```

Comment:
oh, maybe this is the place that saw Kuba ;)
here, we could use MDC to put the params (the scenarioId, version, user, etc) in the context and allow user to use them at the logback configuration level (to make the log less static)